target_snowflake/__init__.py

```diff
@@ -129,8 +130,28 @@ def persist_lines(config, lines, table_cache=None, file_format_type: FileFormatT
             raise Exception(f"Line is missing required key 'type': {line}")
 
         t = o['type']
-
-        if t == 'RECORD':
+        if t == 'FASTSYNC':
+            
+            stream_row_count = 0 
```

Comment:
For organization, it might make sense to move this logic to a fastsync folder that can be called here instead of defining all of the logic in one function.

Reply:
this is how the target currently reads each record.  The new singer row is labeled "FASTSYNC", which relies on the other singer rows for the table (i.e. RECORD, SCHEMA, ACTIVE_VERSION, STATE, etc).  I think to separate it more it would require some extra logic in a few other places. I am fine doing that, but it will be a larger effort of coding/testing. 

This function, in the current state outside this branch, has all the if/else conditions:

```
 if t == 'RECORD':
...
elif t == 'SCHEMA':
....
elif t == 'ACTIVATE_VERSION':
....
elif t == 'STATE':
```

So i threw in the FASTSYNC logic within there, since the table using it will need those other conditions as well in thh same function.   If i am misunderstanding though just let me know and we can chat it out